staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -981,6 +1026,61 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 	return allErrs
 }
 
+func extractMaxElements(schema *apiextensions.JSONSchemaProps) *uint64 {
+	switch schema.Type {
+	case "object":
+		if schema.AdditionalProperties != nil {
+			if schema.MaxProperties != nil {
+				maxProps := uint64(*schema.MaxProperties)
+				return &maxProps
+			}
+			return nil
+		}
+		// return 1, and not nil, because even if we're
+		// not looking at a map here, we can assume the object still preserves
+		// the cardinality of its parents
+		return uint64ptr(1)
+	case "array":
+		if schema.MaxItems != nil {
+			maxItems := uint64(*schema.MaxItems)
+			return &maxItems
+		}
+		return nil
```

Comment:
I'm confused why this is different than the object case above (where we preserve the cardinality of the parent?)

Reply:
The comment might be the source of confusion?  Would it make more sense if it said something like "return 1 to indicate that all fields of an object exist at most one for each occurrence of the object they are fields of" ?  Both the map and array case should return nil if MaxProperties/MaxItems is not present to indicate that the number of children is not bounded.